presto-main/src/main/java/com/facebook/presto/operator/aggregation/QuantileDigestAggregationFunction.java

```diff
@@ -0,0 +1,210 @@
+    public static final QuantileDigestAggregationFunction QDIGEST_AGG_WITH_WEIGHT_AND_ERROR = new QuantileDigestAggregationFunction(parseTypeSignature("V"), parseTypeSignature(StandardTypes.BIGINT), parseTypeSignature(StandardTypes.DOUBLE));
+    public static final String NAME = "qdigest_agg";
+
+    private static final Map<String, MethodHandle> inputFunctions = ImmutableMap.of(
+            StandardTypes.DOUBLE, methodHandle(QuantileDigestAggregationFunction.class, "inputDouble", QuantileDigestState.class, double.class, long.class, double.class),
+            StandardTypes.REAL, methodHandle(QuantileDigestAggregationFunction.class, "inputReal", QuantileDigestState.class, long.class, long.class, double.class),
+            StandardTypes.BIGINT, methodHandle(QuantileDigestAggregationFunction.class, "input", QuantileDigestState.class, long.class, long.class, double.class));
+    private static final MethodHandle COMBINE_FUNCTION = methodHandle(QuantileDigestAggregationFunction.class, "combineState", QuantileDigestState.class, QuantileDigestState.class);
+    private static final MethodHandle OUTPUT_FUNCTION = methodHandle(QuantileDigestAggregationFunction.class, "evaluateFinal", QuantileDigestStateSerializer.class, QuantileDigestState.class, BlockBuilder.class);
+
+    public QuantileDigestAggregationFunction(TypeSignature... typeSignatures)
+    {
+        super(NAME,
+                ImmutableList.of(comparableTypeParameter("V")),
+                ImmutableList.of(),
+                parseTypeSignature("qdigest(V)"),
+                ImmutableList.copyOf(typeSignatures));
+    }
+
+    @Override
+    public String getDescription()
+    {
+        return "Returns a qdigest from the set of reals, bigints or doubles";
+    }
+
+    @Override
+    public InternalAggregationFunction specialize(BoundVariables boundVariables, int arity, TypeManager typeManager, FunctionRegistry functionRegistry)
+    {
+        Type valueType = boundVariables.getTypeVariable("V");
+        QuantileDigestType outputType = (QuantileDigestType) typeManager.getParameterizedType(StandardTypes.QDIGEST,
+                ImmutableList.of(TypeSignatureParameter.of(valueType.getTypeSignature())));
+        return generateAggregation(valueType, outputType, arity);
+    }
+
+    private static InternalAggregationFunction generateAggregation(Type valueType, QuantileDigestType outputType, int arity)
+    {
+        DynamicClassLoader classLoader = new DynamicClassLoader(QuantileDigestAggregationFunction.class.getClassLoader());
+        List<Type> inputTypes = getInputTypes(valueType, arity);
+        QuantileDigestStateSerializer stateSerializer = new QuantileDigestStateSerializer(valueType);
+        Type intermediateType = stateSerializer.getSerializedType();
+
+        checkArgument(inputFunctions.containsKey(valueType.getDisplayName()),
+                "Unsupported type %s supplied, known types: %s", valueType.getDisplayName(), inputFunctions.keySet());
```

Comment:
Declare the 3 method handles are static final fields of this class, and use `switch ... case` to find the appropriate one here. Do not use a map.

This is how code like this have generally been written. There were careful benchmarks. However, I don't remember whether this style is critical to performance.